apps/core/components/userContext/UserContext.js

```diff
@@ -23,9 +41,58 @@ export default class UserContextProvider extends React.Component<Props, State> {
 
     this.state = {
       ...defaultState,
+      actions: {
+        signIn: this.signIn,
+        signOut: this.signOut,
+      },
     };
   }
 
+  componentDidMount() {
+    if (this.state.isFirebaseAvailable) {
+      this.fireBaseListener = firebase.auth().onAuthStateChanged(authUser => {
+        this.setState({
+          isUserSignedIn: authUser != null,
+          ...this.mapUserData(authUser),
+        });
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.fireBaseListener) {
+      /**
+       * NOTE: Auth listener is removed by calling the unsubscribe
+       * function returned by `onAuthStateChanged`
```

Comment:
Maybe add link to Firebase docs for easier reference.

Reply:
Sure, link added.